pkg/config/template/filebased.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package template
+
+import (
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/spf13/afero"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	_ Template = (*FileBasedTemplate)(nil)
+)
+
+// FileBasedTemplate is a JSON Template stored in a file - when it's Template.Content is accessed, that file is read.
+// This is the usual type of Template monaco uses.
+type FileBasedTemplate struct {
+	fs   afero.Fs
+	path string
+}
+
+func (t *FileBasedTemplate) Id() string {
+	return t.path
```

Comment:
This is the total path of the template, but ID is only the file name. Right?

Reply:
In both cases it's a cleaned version of the template's path - that is the combination of the folder the config YAML is in, plus the defined template path from the YAML, with any relative parts resolved. 

ID is really only used in logging and errors in a few places

Reply:
It is not the best name, but we can leave it as it is. As this PR is not about improving the `Template` interface